modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java

```diff
@@ -89,86 +157,468 @@ public IgniteStatisticsRepository statisticsRepository() {
+            GridH2Table table = tableStats.getKey();
+            long rowCount = 0L;
+            Map<String, ColumnStatistics> columnStatistics = new HashMap<>();
+            for (Map.Entry<String, List<ColumnStatistics>> columnStats : tableStats.getValue().entrySet()) {
+                ColumnStatistics globalStatistics = ColumnStatisticsCollector
+                        .aggregate(table::compareValues, columnStats.getValue());
+                if (rowCount < globalStatistics.total())
+                    rowCount = globalStatistics.total();
+                columnStatistics.put(columnStats.getKey(), globalStatistics);
+            }
+
+            ObjectStatisticsImpl objectStatistics = new ObjectStatisticsImpl(rowCount, columnStatistics);
+            statsRepos.saveGlobalStatistics(new StatsKey(table.identifier().schema(), table.identifier().table()),
+                    objectStatistics);
+        }
+        status.doneFut.onDone();
+        */
+
+    }
+
     /**
      * Filter columns by specified names.
      *
-     * @param columns Columns to filter.
+     * @param cols Columns to filter.
      * @param colNames Column names.
      * @return Column with specified names.
      */
-    private Column[] filterColumns(Column[] columns, String... colNames) {
+    private Column[] filterColumns(Column[] cols, @Nullable Collection<String> colNames) {
         if (F.isEmpty(colNames))
-            return columns;
+            return cols;
 
         Set<String> colNamesSet = new HashSet(Arrays.asList(colNames));
-        List<Column> resultList = new ArrayList<>(colNames.length);
+        List<Column> resList = new ArrayList<>(colNames.size());
 
-        for (Column col : columns)
+        for (Column col : cols)
             if (colNamesSet.contains(col.getName()))
-                resultList.add(col);
+                resList.add(col);
 
-        return resultList.toArray(new Column[resultList.size()]);
+        return resList.toArray(new Column[resList.size()]);
     }
 
     /** {@inheritDoc} */
-    @Override public void collectObjectStatistics(String schemaName, String objName, String... colNames)
-            throws IgniteCheckedException {
-        GridH2Table tbl = schemaMgr.dataTable(schemaName, objName);
-        if (tbl == null)
-            throw new IllegalArgumentException(String.format("Can't find table %s.%s", schemaName, objName));
+    @Override public void collectObjectStatistics(
+            String schemaName,
+            String objName,
+            String... colNames
+    ) throws IgniteCheckedException {
+        StatsCollectionFutureAdapter doneFut = new StatsCollectionFutureAdapter();
+        StatsKeyMessage keyMsg = new StatsKeyMessage(schemaName, objName, Arrays.asList(colNames));
+        UUID colId = UUID.randomUUID();
+        StatCollectionStatus status = new StatCollectionStatus(colId, Collections.singletonList(keyMsg),
+            Collections.emptyMap(), doneFut);
```

Comment:
What is 'doneFut' future purpose? Why we pass a future to status object constructor from outside?

Reply:
It need to async collection and to cancel started collection. But yes, no reason to pass it from the outside. Fixed.